### PR TITLE
890 remove useAsyncEffect

### PR DESCRIPTION
**Pull request by dhaneshnm**
Removes useAsyncEffect from DashboardList, StudioContainer and
WorkspaceList.

Fixes #BlueBrain/nexus/issues/890

### CODE REVIEW

File: src/shared/containers/DashboardListContainer.tsx [hunk 1]
```diff
@@ -29,22 +28,23 @@ const DashboardList: React.FunctionComponent<DashboardListProps> = ({
...
+          encodeURIComponent(dashboardObject.dashboard)
+        );
+      })
+    )
+      .then(values => {
+        setDashboardResources(values);
+        setSelectedDashboard(values[0]);
+      })
+      .catch(e => {
+        // Fail silently.
```
Comment (Stafeeva) [inline src/shared/containers/DashboardListContainer.tsx]: we can use andt notification API here to display an error to a user, for example something like this:
```
        notification.error({
          message: `An error occurred`,
          description: error.message,
        });
```

File: src/shared/containers/StudioContainer.tsx [hunk 2]
```diff
@@ -27,17 +26,19 @@ const StudioContainer: React.FunctionComponent<StudioContainerProps> = ({
...
+      .then(value => {
+        if (value['@type'] === 'Studio') {
+          const studioResource: StudioResource = value as StudioResource;
+          setStudioResource(studioResource);
+          const workspaceIds: string[] = studioResource['workspaces'];
+          setWorkspaceIds(workspaceIds);
+        }
+      })
+      .catch(e => {
+        // Fail silently.
```
Comment (Stafeeva) [inline src/shared/containers/StudioContainer.tsx]: Same as above

File: src/shared/containers/WorkspaceListContainer.tsx [hunk 3]
```diff
@@ -23,21 +22,24 @@ const WorkspaceList: React.FunctionComponent<WorkspaceListProps> = ({
...
+          path: workspaceId,
+        });
+      })
+    )
+      .then(values => {
+        setWorkspaces(values);
+        setSelectedWorkspace(values[0] as Resource);
+      })
+      .catch(e => {
+        // Fail silently.
```
Comment (Stafeeva) [inline src/shared/containers/WorkspaceListContainer.tsx]: Please :)

File: src/shared/containers/DashboardListContainer.tsx (hunk 1 above)
Comment (dhaneshnm) [reply src/shared/containers/DashboardListContainer.tsx]: It was decided earlier that we are going to come back to this later. We have not decided what error to display to the user yet.

**Pull request merged**